### ISSUE TITLE
Hyper-V Dynamic Memory `hv_balloon: Memory hot add failed`

### ISSUE THREAD

**Issue by CMCDragonkai** [NixOS/nix#1276]
I'm trying out NixOS on Hyper-V.

I noticed while trying to install NixOS onto a new Hyper-V VM (I used the 17.03beta347.f51f167 ISO), that if the Hyper-V VM has dynamic memory enabled, when trying to install something using `nix-env -i git` on the booted ISO, we get kernel errors like: `hv_balloon: Memory hot add failed`. The error is reported immediately and then repeatedly while the command is running. After running for a couple minutes, I cancelled it.

Basically it appears Hyper-V memory ballooning is not supported by the NixOS ISO.

You can try it by setting starting RAM to 512 MiB, then enable dynamic memory between 512 MiB and 2048 MiB.

I found that running `nix-env -i git` requires at least 1.5 GiB of RAM for it succeed.

**Comment by copumpkin** [comment]
This sounds like a nixpkgs issue, if it's the NixOS kernel. I doubt anyone relevant will see it here.

**Comment by FossPrime** [comment]
I got this error while installing NixOS. Turns out I was impatient with disk destroyer. The time that it looked like it worked, it just made a file in /dev. Etcher and USBImager weren't available in the Debian/raspberry Pi store. Gparted lacked the flashing feature. gnome-disk-management was in the repositories, and did the trick.
